src/Forms/GridField/GridField.php

```diff
@@ -290,6 +298,18 @@ public function setConfig(GridFieldConfig $config)
         return $this;
     }
 
+    /**
+     * @param bool $readonly
+     *
+     * @return $this
+     */
+    public function setReadonly($readonly)
+    {
+        parent::setReadonly($readonly);
+        $this->getState()->Readonly($readonly);
```

Comment:
Wondering, only because this is so ugly and IDE unfriendly, is this functionally equivalent to just `getState()->Readonly = $readonly`? I think it's just a question of whether it uses `__call()` or `__set()`.

Reply:
Done